packages/@ama-sdk/core/src/fwk/api.helpers.ts

```diff
@@ -115,3 +116,45 @@ export function tokenizeRequestOptions(tokenizedUrl: string, queryParameters: {
 
   return {values, url: tokenizedUrl, queryParams: tokenizedQueryParams};
 }
+
+/**
+ * Compute the reviver to use in case the success response code returned by the API does not match the API specifications.
+ * Does not try to match non-successful responses as error are handled separately
+ *
+ * @param revivers
+ * @param response
+ * @param disableFallback
+ */
+export function getResponseReviver<T>(revivers: {[statusCode: number]: ReviverType<T> | undefined} | undefined | ReviverType<T>,
+                                                          response: {status?: number; ok?: boolean} | undefined, disableFallback?: boolean):
+  {reviver: ReviverType<T> | undefined; logMessage: string | undefined} {
```

Comment:
Having a optional logger in input would be simpler and would not require the creation of an object in the function return.
But why not :)